src/variable_explorer/variable_explorer_helpers.py

```diff
@@ -56,6 +56,13 @@ def get_categories(np_array):
 
     return [{"name": name, "count": count} for name, count in categories]
 
+# Cast potentially non json-serializable objects to strings to avoid ValueErrors during iPython json serialization
+def cast_objects_to_string(df):
+    for column in df:
+        # List of types: object, int64, float64, bool, datetime64m timedelta[ns], category (https://pbpython.com/pandas_dtypes.html)
+        if df[column].dtype == 'object':
+            df[column] = df[column].apply(str)
```

Comment:
Maybe [astype](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.astype.html) would have better performance?

```suggestion
            df[column] = df[column].astype(str)
```

Reply:
I can't right now, but can you please check if this doesn't break the column statistics? In each column header we show you whether the column is an integer, string, np.array, etc... To make things worse, a lot of dtypes have subtypes. I am worried that if we cast it as a string it might break some of this fragile logic.

Reply:
I considered that. We should be fine, because this method is called at very end – after all the info you mention has already been gathered.

Reply:
I moved the casting to the end of the function just to make sure there is no mutation I am not aware of. Also, I've added a simple test case for dtype send to frontend.

@the21st Regarding the performance, I did a quick search when I was writing this and apply seemed like a better option, but I'm quite noob in pandas so I'm open to change it.

Reply:
Did a [quick experiment](https://deepnote.com/project/ba30b46f-29dc-4bea-ae27-b96e527c4871) and they came out pretty much the same, so feel free to keep as is.